vendor/github.com/Azure/azure-sdk-for-go/services/web/mgmt/2021-02-01/web/models.go

```diff
@@ -24464,6 +24464,8 @@ type SitePatchResourceProperties struct {
 	// VirtualNetworkSubnetID - Azure Resource Manager ID of the Virtual network and subnet to be joined by Regional VNET Integration.
 	// This must be of the form /subscriptions/{subscriptionName}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{vnetName}/subnets/{subnetName}
 	VirtualNetworkSubnetID *string `json:"virtualNetworkSubnetId,omitempty"`
+	// PublicNetworkAccess - Property to allow or block all public traffic.
+	PublicNetworkAccess *string `json:"publicNetworkAccess,omitempty"`
```

Comment:
we shouldn't be changing any vendored files? is this not availible in the current SDK then?

Reply:
Hi Katbyte,

Not sure if you saw my message above calling this out so I'll put it below.

I'm seeing some unit test failures as I have updated the Models for web against version 2021, the field 'publicNetworkAccess' does not exist against the Azure API for this API version 2021-02-01 (it only exists in the site.config block). Newer versions of the API support this field.


- What's the process here to update the model version?
- I do understand why we would want to verify the models but why are we doing this if Terraform does not take into account the explicit API version when it calls the API? This has been proven by the successful calls I've made against Azure with these changes in place.

Looking for some feedback and support here, thanks!!

Reply:
Hi @katbyte,

Not sure if you saw my message above calling this out so I'll put it below.

I'm seeing some unit test failures as I have updated the Models for web against version 2021, the field 'publicNetworkAccess' does not exist against the Azure API for this API version 2021-02-01 (it only exists in the site.config block). Newer versions of the API support this field.

- What's the process here to update the model version?
- I do understand why we would want to verify the models but why are we doing this if Terraform does not take into account the explicit API version when it calls the API? This has been proven by the successful calls I've made against Azure with these changes in place.

Looking for some feedback and support here, thanks!!

Reply:
sorry i missed that message when reviewing - so this resource is still using the old track 1 SDK and to make use of the newer APIs/SDKs this resource will need to by migrated over to go-azure-sdk - currently 2023-01-01 is already vendor'd into the provider already. If a newer version is required it would need to be added to the list of generated SDKs here: https://github.com/hashicorp/pandora/blob/main/config/resource-manager.hcl.

It should be pretty straight forward to do this migration as the newer sdk is pretty close to the old one. An example of this can be found in this PR: https://github.com/hashicorp/terraform-provider-azurerm/pull/24760 which is also migrating a resource using the web api.

Reply:
thanks for the detailed response @katbyte ,
Do you know if this change is already on a roadmap somewhere at Hashicorp or is this migration/enhancement low priority?

Reply:
thanks for the detailed response @katbyte ,
Do you know if this change is already on a roadmap somewhere at Hashicorp or is this migration/enhancement low priority?

Reply:
@galbantow migrating all resources possible to the newer go-azure-sdk is on our roadmap but low priority, we are doing as needed and as we have time (out of 1300 resources and data sources i believe only 100-200 are left) 

as for adding site properties public network access that is not currently on a roadmap, but as there is this PR open it will remain on our radar till done & just become part of the work to migrate the resource over.